policyengine_us/variables/gov/states/mt/tax/income/mt_income_tax_before_refundable_credits.py

```diff
@@ -0,0 +1,35 @@
+from policyengine_us.model_api import *
+
+
+class mt_income_tax_before_refundable_credits(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Montanta income tax before refunable credits"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        income = tax_unit("mt_taxable_income", period)
+        p = parameters(period).gov.states.mt.tax.income.main
+        # Before 2024 Montana has a flat tax rate for each filing status
+        temporary_rate = p.temporary_rate.calc(income)
```

Comment:
More verbose but I'd suggest putting the non-filing-status-specific rates in the filing status files to make the Python simpler.